### PR TITLE
Declaring Java interface deprecation

### PR DESCRIPTION
**Pull request by napetrov**
# Description
This PR declare deprecation of java interfaces in oneDAL

### CODE REVIEW

File: INSTALL.md [hunk 1]
```diff
@@ -60,7 +60,8 @@ For details, see [System Requirements for oneDAL](https://software.intel.com/con
 
             call "C:\Program Files (x86)\Intel\oneAPI\compiler\latest\env\vars.bat"
 
-4. Set the environment variables for one of the supported Java\* compilers. For example:
+4. Deprecation Notice: Java interfaces in the oneDAL library have been deprecated and may no longer be supported in future releases.
```
Comment (alexandraepan) [inline INSTALL.md:63]:
```suggestion
4. Deprecation Notice: The Java interfaces are deprecated in the oneDAL library and may no longer be supported in future releases.
```

File: README.md [hunk 2]
```diff
@@ -34,6 +34,7 @@ There are different ways for you to build high-performance data science applicat
 - Use [Intel(R) Extension for Scikit-learn*](https://intel.github.io/scikit-learn-intelex/) to accelerate existing scikit-learn code without changing it
 - Use [daal4py](https://github.com/intel/scikit-learn-intelex/tree/master/daal4py), a standalone package with Python API for oneDAL
 - Use [Scala and Java interfaces](#apache-spark-mllib) that match Apache Spark MlLib API
+Deprecation Notice: Java interfaces in the oneDAL library have been deprecated and may no longer be supported in future releases.
```
Comment (alexandraepan) [inline README.md:37]:
```suggestion
Deprecation Notice: The Java interfaces are deprecated in the oneDAL library and may no longer be supported in future releases.
```

File: README.md [hunk 3]
```diff
@@ -55,15 +56,16 @@ Python Examples:
 
 <details><summary>Other Examples</summary>
 
+- [MPI](https://github.com/oneapi-src/oneDAL/tree/master/samples/daal/cpp/mpi)
+- [MySQL](https://github.com/oneapi-src/oneDAL/tree/master/samples/daal/cpp/mysql)
+Deprecation Notice: Java interfaces in the oneDAL library have been deprecated and may no longer be supported in future releases. This includes Scala, Spark and Hadoop samples
```
Comment (alexandraepan) [inline README.md:61]:
```suggestion
Deprecation Notice: The Java interfaces are deprecated in the oneDAL library and may no longer be supported in future releases. This includes Scala, Spark, and Hadoop samples.
```

File: docs/source/daal-interfaces.rst [hunk 4]
```diff
@@ -39,5 +39,5 @@ Examples
 You can find examples on Github*:
 
 - `C++ <https://github.com/oneapi-src/oneDAL/tree/master/examples/daal/cpp/source>`_ (CPU)
-- `Java* <https://github.com/oneapi-src/oneDAL/tree/master/examples/daal/java/com/intel/daal/examples>`_ (not supported on GPU)
+- `Java* <https://github.com/oneapi-src/oneDAL/tree/master/examples/daal/java/com/intel/daal/examples>`_ (nDeprecation Notice: Java interfaces in the oneDAL library have been deprecated and may no longer be supported in future releases.)
```
Comment (alexandraepan) [inline docs/source/daal-interfaces.rst:42]:
```suggestion
- `Java* <https://github.com/oneapi-src/oneDAL/tree/master/examples/daal/java/com/intel/daal/examples>`_ (Deprecation Notice: The Java interfaces are deprecated in the oneDAL library and may no longer be supported in future releases.)
```

File: docs/source/daal/algorithms/association_rules/association-rules.rst [hunk 5]
```diff
@@ -234,6 +234,7 @@ Examples
   .. tab:: Java*
 
     .. note:: There is no support for Java on GPU.
+    .. note:: Java interfaces in the oneDAL library have been deprecated and may no longer be supported in future releases.
```
[the remark this answers is not in the document]
Comment (alexandraepan) [reply docs/source/daal/algorithms/association_rules/association-rules.rst:237]: It's the same for other places

File: docs/source/daal/algorithms/decision_forest/decision-forest-classification.rst [hunk 6]
```diff
@@ -160,7 +160,8 @@ Examples
 
   .. tab:: Java*
 
-    .. note:: There is no support for Java on GPU.
+   .. note:: There is no support for Java on GPU.
```
Comment (alexandraepan) [inline docs/source/daal/algorithms/decision_forest/decision-forest-classification.rst:163]: Maybe it is worth making this note just a sentence? It worked before, but now two notes are not the best way to present the info

Comment (napetrov) [reply docs/source/daal/algorithms/decision_forest/decision-forest-classification.rst]: think worth deleting this altogether assuming deprecation

**Review by alexandraepan — APPROVED** [review]

**Pull request merged**